src/ICSharpCode.SharpZipLib/Zip/ZipFile.cs

```diff
@@ -2511,6 +2513,15 @@ private int FindExistingUpdate(string fileName)
 			return result;
 		}
 
+		private bool IsDirectoryName(string name)
```

Comment:
`GetTransformedFileName` is only referenced in `FindExistingUpdate`, perhaps `GetTransformedFileName`, `GetTransformedDirectoryName` and `IsDirectoryName` should be reduced to a single function called `GetTransformedEntryName`? The two former ones are almost identical.